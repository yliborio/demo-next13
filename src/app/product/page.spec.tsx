import products from "../../mocks/products.json";
import { render, screen, waitFor } from "@testing-library/react";
import Page from "./page";

jest.mock("next/navigation");

describe("Page", () => {
  beforeAll(() => {
    global.fetch = () =>
      Promise.resolve({
        json: () => Promise.resolve(products[0]),
      } as any);
  });
  it("should fetch and render a product", async () => {
    const page = await Page({ searchParams: { id: "1" } });
    const { asFragment } = render(page);

    expect(asFragment()).toMatchSnapshot();
    const productTitle = screen.getByText(products[0].title);
    expect(productTitle).toBeInTheDocument();
  });
});
