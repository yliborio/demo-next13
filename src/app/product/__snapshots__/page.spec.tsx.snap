// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Page should fetch and render a product 1`] = `
<DocumentFragment>
  <div
    class="container"
  >
    <img
      alt=""
      class="image"
      data-nimg="1"
      decoding="async"
      height="0"
      loading="lazy"
      sizes="100vw"
      src="/_next/image?url=https%3A%2F%2Ffakestoreapi.com%2Fimg%2F81fPKd-2AYL._AC_SL1500_.jpg&w=3840&q=75"
      srcset="/_next/image?url=https%3A%2F%2Ffakestoreapi.com%2Fimg%2F81fPKd-2AYL._AC_SL1500_.jpg&w=640&q=75 640w, /_next/image?url=https%3A%2F%2Ffakestoreapi.com%2Fimg%2F81fPKd-2AYL._AC_SL1500_.jpg&w=750&q=75 750w, /_next/image?url=https%3A%2F%2Ffakestoreapi.com%2Fimg%2F81fPKd-2AYL._AC_SL1500_.jpg&w=828&q=75 828w, /_next/image?url=https%3A%2F%2Ffakestoreapi.com%2Fimg%2F81fPKd-2AYL._AC_SL1500_.jpg&w=1080&q=75 1080w, /_next/image?url=https%3A%2F%2Ffakestoreapi.com%2Fimg%2F81fPKd-2AYL._AC_SL1500_.jpg&w=1200&q=75 1200w, /_next/image?url=https%3A%2F%2Ffakestoreapi.com%2Fimg%2F81fPKd-2AYL._AC_SL1500_.jpg&w=1920&q=75 1920w, /_next/image?url=https%3A%2F%2Ffakestoreapi.com%2Fimg%2F81fPKd-2AYL._AC_SL1500_.jpg&w=2048&q=75 2048w, /_next/image?url=https%3A%2F%2Ffakestoreapi.com%2Fimg%2F81fPKd-2AYL._AC_SL1500_.jpg&w=3840&q=75 3840w"
      style="color: transparent;"
      width="0"
    />
    <div
      class="info"
    >
      <div
        class="title"
      >
        Fjallraven - Foldsack No. 1 Backpack, Fits 15 Laptops
      </div>
      <span
        class="description"
      >
        Description:
      </span>
      <div
        class="content"
      >
        Your perfect pack for everyday use and walks in the forest. Stash your laptop (up to 15 inches) in the padded sleeve, your everyday
      </div>
      <div
        class="action"
      >
        <span
          class="price"
        >
          $109.95 
        </span>
        <button
          class="button"
        >
          <span>
            Add to Cart
          </span>
          <svg
            fill="white"
            height="32"
            id="cart"
            viewBox="0 0 32.402 32"
            width="32.402"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M6 30a2 2 1080 1 0 4 0 2 2 1080 1 0-4 0zm18 0a2 2 1080 1 0 4 0 2 2 1080 1 0-4 0zM-.058 5a1 1 0 0 0 1 1H3.02l1.242 5.312L6 20c0 .072.034.134.042.204l-1.018 4.58A.997.997 0 0 0 6 26h22.688a1 1 0 0 0 0-2H7.248l.458-2.06c.1.016.19.06.294.06h18.23c1.104 0 1.77-.218 2.302-1.5l3.248-9.964C32.344 8.75 31.106 8 30 8H6c-.156 0-.292.054-.438.088l-.776-3.316A1 1 0 0 0 3.812 4H.942a1 1 0 0 0-1 1zm6.098 5h23.81l-3.192 9.798c-.038.086-.07.148-.094.19-.066.006-.17.012-.334.012H8v-.198l-.038-.194L6.04 10z"
            />
          </svg>
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
